backendservice/src/main/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepository.java

```diff
@@ -79,18 +80,37 @@ public UUID createShop(CreateShopRequest createShopRequest) {
 		}
 
 		// TODO(b/158975796): Simplify logging across all backend APIs
-		Logger.getLogger(InMemoryVirtualQueueRepository.class.getName()).log(Level.SEVERE, "Tried to fetch tokens of a shop which is not in the ACTIVE state.");
+		Logger.getLogger(InMemoryVirtualQueueRepository.class.getName()).log(Level.SEVERE,
+				"Tried to fetch tokens of a shop which is not in the ACTIVE state.");
 		return new ArrayList<>();
 	}
 
-	// Method returns all shops keeping in mind the feature of restoring shops later on.
+	public Token getNewToken(UUID shopId) {
+		if (shopMap.get(shopId).getStatus() == ShopStatus.ACTIVE) {
+			Integer newTokenNumber = shopIdToLastAllotedTokenMap.get(shopId).incrementAndGet();
+			UUID uuid = UUID.randomUUID();
+			Token newToken = new Token(uuid, shopId, newTokenNumber);
+			newToken.setStatus(Token.Status.ACTIVE);
+			tokenMap.put(uuid, newToken);
+			List<Token> newTokenList = shopIdToListOfTokensMap.get(shopId);
```

Comment:
Nit. Can we try to do this in one statement itself? eg:
shopIdToListOfTokensMap.get(shopId).add(newToken)

Reply:
Yes done